src/main/java/qna/domain/Question.java

```diff
@@ -85,8 +85,27 @@ public boolean isDeleted() {
 		return deleted;
 	}
 
-	public void setDeleted(boolean deleted) {
-		this.deleted = deleted;
+	public List<DeleteHistory> delete(User loginUser) throws CannotDeleteException {
+		if (!this.canDeleteQuestion(loginUser)) {
+			throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
+		}
+		setDelete();
+
+		return addDeleteHistories(loginUser);
+	}
+
+	private List<DeleteHistory> addDeleteHistories(User loginUser) {
+		List<DeleteHistory> deleteHistories = new ArrayList<>();
+		deleteHistories.add(new DeleteHistory(ContentType.QUESTION, this.getId(), loginUser));
+
+		this.answers.makeDeleteHistories(loginUser)
+			.stream()
+			.forEach(deleteHistory -> deleteHistories.add(deleteHistory));
+		return deleteHistories;
+	}
```

Comment:
다음과 같이 해볼 수 있을 것 같아요! 😄 
```suggestion
	private List<DeleteHistory> addDeleteHistories(User loginUser) {
		List<DeleteHistory> deleteHistories = new ArrayList<>();
		deleteHistories.add(new DeleteHistory(ContentType.QUESTION, this.getId(), loginUser));
		deleteHistories.addAll(this.answers.makeDeleteHistories(loginUser));
		return deleteHistories;
	}
```